test/modules/elevationProfile/components/ElevationProfile.test.js

```diff
@@ -836,4 +836,29 @@ describe('ElevationProfile', () => {
 	});
 
 
+	describe('when disconnected', () => {
+		it('destroys chart', async () => {
+			// arrange
+			const coordinates = [
+				[0, 1],
+				[2, 3]
+			];
+
+			spyOn(elevationServiceMock, 'getProfile').withArgs(coordinates).and.resolveTo(profile());
+			const element = await setup({
+				elevationProfile: {
+					active: true,
+					coordinates: coordinates
+				}
+			});
+
+			const onDisconnectSpy = spyOn(element, 'onDisconnect').and.callThrough();
+
+			//act
+			element.onDisconnect(); // we have to call onDisconnect manually
+
+			// assert
+			expect(onDisconnectSpy).toHaveBeenCalled();
```

Comment:
Hmm, this setup doesn't make sense.
We want to test if the observer's unsubscribe function has been called.
Have a look at the OlMap's test:
https://github.com/ldbv-by/bav4-nomigration/blob/90785ff4182311b32587031396fa398f01c59af6/test/modules/olMap/components/OlMap.test.js#L271